### DISCUSSION TITLE
Suggestion to rename "cancelAll" and "withdrawAll"

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/evm-monorepo#212] [Ideas]
The current names might be misconstrued as meaning to meaning to "withdraw everything". Wouldn't `withdrawMultiple` be a better name?

**Comment by andreivladbrg** [comment]
`withdrawMultiple` would work for me

**Reply by PaulRBerg** [reply]
Thanks. @razgraf your input would be helpful here.

**Reply by PaulRBerg** [reply]
Now that I think about it, `withdrawAll` is also confusing because it could be interpreted as meaning to withdraw from all the streams that the user has. Which is of course not possible in the core cotnracts because we don't index streams by sender or recipient address.

**Reply by razgraf** [reply]
The fact that `withdrawAll` is accompanied by a parameter describing an array of ids makes it pretty clear it will cause a "withdraw action on all given ids". I've seen this pattern used before so I'm not sure we need to change names here.

**Reply by PaulRBerg** [reply]
You don't always see the array of ids "accompanying" the `withdrawAll` name. I think it would be quite useful to rename it to something that better indicates a plurality of streams, so as not to confuse it with:

1. Maximum withdrawable amount
2. Withdraw all streams that belong to the user

I think it's really easy to mistake `withdrawAll` for one of the purposes listed above. The potential confusion risk is not worth it.

**Reply by razgraf** [reply]
> Maximum withdrawable amount [vs] withdraw all streams

True, this actually makes it prone to confusion.

Ideas: `withdrawMultiple` (Andrei's), `withdrawList`, `withdrawCollection`, `withdrawMore` / `withdrawMany`

**Comment by PaulRBerg** [comment]
Another idea would be to overload `withdraw`, but as per our discussions regarding the create functions, overload are not super use-friendly. Lots of ecosystem tools (like Ethers) are having a hard time disambiguating between the overloaded functions, which leads to a sub-par developer experience.

**Comment by PaulRBerg** [comment]
Locking, implemented in #220 (`withdrawMultiple` and `cancelMultiple`).
